src/display.c

```diff
@@ -420,7 +441,11 @@ PyImaging_GrabScreenWin32(PyObject *self, PyObject *args) {
     PyErr_SetString(PyExc_OSError, "screen grab failed");
 
     DeleteDC(screen_copy);
-    DeleteDC(screen);
+    if (screens == -1) {
+        ReleaseDC(wnd, screen);
```

Comment:
You added this for the `error:` branch but not for the success branch.

Reply:
Ok, I've updated the commit to add it or success as well.